terraform/aws/kube/kube.tf

```diff
@@ -0,0 +1,202 @@
+  description = "The name of the file storage bucket."
+  type        = string
+}
+
+variable "file_storage_bucket_region" {
+  description = "The region in which the file storage bucket resides."
+  type        = string
+}
+
+variable "file_metadata_queue_name" {
+  description = "The name of the file metadata queue."
+  type        = string
+}
+
+variable "database_endpoint" {
+  description = "The endpoint for the database."
+  type        = string
+}
+
+##########################################
+# Kubernetes
+##########################################
+
+resource "kubernetes_deployment" "wandb" {
+  metadata {
+    name = "wandb"
+    labels = {
+      app = "wandb"
+    }
+  }
+
+  spec {
+    strategy {
+      type = "RollingUpdate"
+    }
+
+    replicas = 1
+
+    selector {
+      match_labels = {
+        app = "wandb"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = "wandb"
+        }
+      }
+
+      spec {
+        container {
+          name              = "wandb"
+          image             = "wandb/local:${var.wandb_version}"
+          image_pull_policy = "Always"
+
+          env {
+            name  = "LICENSE"
+            value = var.license
+          }
+          env {
+            name  = "BUCKET"
+            value = "s3:///${var.file_storage_bucket_name}"
+          }
+          env {
+            name  = "BUCKET_QUEUE"
+            value = "sqs://${var.file_metadata_queue_name}"
+          }
+          env {
+            name  = "AWS_REGION"
+            value = var.file_storage_bucket_region
+          }
+          env {
+            name  = "MYSQL"
+            value = "mysql://wandb:wandb_root_password@${var.database_endpoint}/wandb_local"
+          }
+
+          port {
+            name           = "http"
+            container_port = 8080
+            protocol       = "TCP"
+          }
+
+          liveness_probe {
+            http_get {
+              path = "/healthz"
+              port = "http"
+            }
+          }
+          readiness_probe {
+            http_get {
+              path = "/ready"
+              port = "http"
+            }
+          }
+
+          resources {
+            requests {
+              cpu    = "1500m"
+              memory = "4G"
+            }
+            limits {
+              cpu    = "4000m"
+              memory = "8G"
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+resource "kubernetes_service" "wandb_service" {
+  metadata {
+    name = "wandb"
+  }
+
+  spec {
+    type = "NodePort"
+    selector = {
+      app = "wandb"
+    }
+    port {
+      port      = 8080
+      node_port = 32543
+    }
+  }
+}
+
+resource "local_file" "kubeconfig" {
+  filename = "kubeconfig.yaml"
+  content  = <<KUBECONFIG
+    apiVersion: v1
+    kind: Config
+    clusters:
+    - cluster:
+        server: ${var.kube_cluster_endpoint}
+        certificate-authority-data: ${var.kube_cert_data}
```

Comment:
Ahh, I saw this and thought it was SSL, this is just for the kubectl client, I see.